src/noise-models.lisp

```diff
@@ -0,0 +1,271 @@
+;;;; noise-models.lisp
+;;;;
+;;;; Author: Sophia Ponte
+
+(in-package #:qvm)
+
+;;; Implementation of rule-based noise models for the QVM.
+
+;;; These values are relatively aribtrary limits for NOISE-PREDICATE
+;;; priorty. They are used to order noise models for matching to
+;;; instructions.
+(defconstant +MAX-PRIORITY+ 20)
+(defconstant +MIN-PRIORITY+ 0)
+(alexandria:define-constant +DEFAULT-NOISE-PREDICATE-NAME+ "NOISE-NAME" :test #'string=)
+
+
+;;; A noise predicate is a function that takes an instruction's gate
+;;; and qubits, and returns either true if the instruction is a match,
+;;; else false. The priority indicates the priority of the predicate
+;;; (tbd) and the noise-position indicates whether the predicate
+;;; should match before or after an instruction.
+(defclass noise-predicate ()
+  ((predicate-function
+    :initarg :predicate-function
+    :reader predicate-function
+    :documentation "Does the instruction trigger the application of a channel?")
+   (priority
+    :initarg :priority
```

Comment:
Suggestion: create a `noise-priority` type that encompasses integers from `+min-priority+` to `+max-priority+`.